### PR TITLE
STAC-14046 Kubelet unverified TLS fallback

### PR DESCRIPTION
**Pull request by julianosk**
### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test
- [ ] E2E/Molecule test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Did you add release notes describing the changes you made?
- [ ] Yes

### Step 6: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: pkg/util/kubernetes/kubelet/kubelet.go [hunk 1]
```diff
@@ -749,27 +785,46 @@ func (ku *KubeUtil) setKubeletHost(hosts *connectionInfo, httpsPort, httpPort in
...
 	}
 
 	return "", connectionErrors
 }
 
+func tryKubeletConnection(ku *KubeUtil, host string, httpsPort int, verifyTLS bool) error {
+	ku.kubeletHost = host
+	err := ku.setupKubeletAPIClient(verifyTLS)
+	if err != nil {
+		log.Debugf("Cannot setup https kubelet api client for %s: %v. TLS verification: %v.", host, err, verifyTLS)
```
Comment (Q69K) [inline pkg/util/kubernetes/kubelet/kubelet.go:815]:
```suggestion
		log.Debugf("Cannot setup https kubelet api client for %s:%d: %v. TLS verification: %v.", host, httpsPort, err, verifyTLS)
```

File: pkg/util/kubernetes/kubelet/kubelet.go [hunk 2]
```diff
@@ -749,27 +785,46 @@ func (ku *KubeUtil) setKubeletHost(hosts *connectionInfo, httpsPort, httpPort in
...
+		return err
+	}
+
+	err = checkKubeletHTTPSConnection(ku, httpsPort)
+	if err == nil {
+		log.Debugf("Can connect to kubelet using %s and HTTPS. TLS verification: %v.", host, verifyTLS)
```
Comment (Q69K) [inline pkg/util/kubernetes/kubelet/kubelet.go:821]:
```suggestion
		log.Debugf("Can connect to kubelet using %s:%d and HTTPS. TLS verification: %v.", host, httpsPort, verifyTLS)
```

File: pkg/util/kubernetes/kubelet/kubelet.go [hunk 3]
```diff
@@ -749,27 +785,46 @@ func (ku *KubeUtil) setKubeletHost(hosts *connectionInfo, httpsPort, httpPort in
...
+		return nil
+	}
+	log.Debugf("Cannot connect to kubelet using %s and https: %v. TLS verification: %v.", host, err, verifyTLS)
```
Comment (Q69K) [inline pkg/util/kubernetes/kubelet/kubelet.go:824]:
```suggestion
	log.Debugf("Cannot connect to kubelet using %s:%d and https: %v. TLS verification: %v.", host, httpsPort, err, verifyTLS)
```

File: pkg/util/kubernetes/kubelet/kubelet.go (hunk 2 above)
Comment (Q69K) [reply pkg/util/kubernetes/kubelet/kubelet.go:821]: i'd use some more positive sentence)) like "Connection to kubelet is successful"

**Review by Q69K — APPROVED** [review]

**Review by julianosk — COMMENTED** [review]
I'll refactor to make it easier to do upstream merges in the future

**Review by zandrewitte — APPROVED** [review]

**Pull request merged**